core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitorSpec.scala

```diff
@@ -1,10 +1,109 @@
+    )
+    val job1 = new WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      JobType.AzureLandingZoneResult,
+      None,
+      Some("bpId1"),
+      Timestamp.from(Instant.now())
+    )
+    val jobDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    when(jobDao.selectAll()).thenReturn(Future.successful(Seq(job0, job1)))
+    val monitor = spy(new WorkspaceResourceMonitor(jobDao, List()))
+    doReturn(Future.successful(true)).when(monitor).runJob(ArgumentMatchers.eq(job0))
+    doReturn(Future.successful(false)).when(monitor).runJob(ArgumentMatchers.eq(job1))
+
+    Await.result(monitor.checkJobs(), Duration.Inf) shouldBe CheckDone(1)
+  }
+
+  behavior of "WorkspaceResourceMonitor.runJob"
+
+  it should "delete a job after it completes successfully" in {
+    val runner = spy(new WorkspaceManagerResourceJobRunner {
+      override val jobType: JobType = JobType.AzureLandingZoneResult
+      override def run(job: WorkspaceManagerResourceMonitorRecord)(implicit
+        executionContext: ExecutionContext
+      ): Future[Boolean] = Future.successful(true)
+    })
+    val job = new WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      JobType.AzureLandingZoneResult,
+      None,
+      Some("bpId"),
+      Timestamp.from(Instant.now())
+    )
+    val jobDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    when(jobDao.selectAll()).thenReturn(Future.successful(Seq(job)))
+    when(jobDao.delete(ArgumentMatchers.any())).thenReturn(Future.successful(true))
+    val monitor = new WorkspaceResourceMonitor(jobDao, List(runner))
 
-class WorkspaceResourceMonitorSpec extends AnyFlatSpec {
+    Await.result(monitor.runJob(job), Duration.Inf) shouldBe true
+    verify(jobDao).delete(ArgumentMatchers.any())
 
+  }
 
+  it should "call all job runners registered for a job type" in {
+    val runner0 = spy(new WorkspaceManagerResourceJobRunner {
+      override val jobType: JobType = JobType.AzureLandingZoneResult
+      override def run(job: WorkspaceManagerResourceMonitorRecord)(implicit
+        executionContext: ExecutionContext
+      ): Future[Boolean] = Future.successful(true)
+    })
+    val runner1 = spy(new WorkspaceManagerResourceJobRunner {
+      override val jobType: JobType = JobType.AzureLandingZoneResult
+      override def run(job: WorkspaceManagerResourceMonitorRecord)(implicit
+        executionContext: ExecutionContext
+      ): Future[Boolean] = Future.successful(false)
+    })
+    val job = new WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      JobType.AzureLandingZoneResult,
+      None,
+      Some("bpId"),
+      Timestamp.from(Instant.now())
+    )
+    val jobDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    when(jobDao.selectAll()).thenReturn(Future.successful(Seq(job)))
+    val monitor = new WorkspaceResourceMonitor(jobDao, List(runner0, runner1))
 
+    Await.result(monitor.runJob(job), Duration.Inf) shouldBe false
```

Comment:
Does it return false because the second job returns false?

Reply:
I see now that the runner returns returns false if the job isn't done, so I think I answered my own question.

Reply:
yes (any of them, really)